docs/docsite/rst/testing_modules.rst

```diff
@@ -0,0 +1,196 @@
+          description: Ansible tenant
+          state: present
+
+      - name: Add a new parent object
+        aci_l3out:
+          <<: *aci_info
+          tenant: ansible_tenant
+          name: '<parent_object_name>' # Replace <parent_object_name> with the actual name of the parent object, e.g., ansible_l3out
+          description: Parent object description
+          state: present
+
+
+.. code-block:: text
+
+6. The next section consists of adding tasks for all aspects of our module. We include Ansible's register attribute to save the result of the task. The procedure is as follows:
+
+* Tasks to clean the environment to begin with a clean slate.
+        * Task to add a new object with only the required parameters. (check mode) -> where the check_mode is set to true.
+        * Task to add a new object with only the required parameters. (actual run) -> where the check_mode is set to false.
+        * task to add a new object with only the required parameters again. (idempotency), to test the idempotency of the module.
+        * An assert task to verify the object was created successfully for the previous tasks of adding new object with only the required parameters.(check mode, actual run and idempotency).
+        * Task to add a new object with all the parameters. (check mode) -> where the check_mode is set to true.
+        * Task to add a new object with all the parameters. (actual run) -> where the check_mode is set to false.
+        * Task to add a new object with all the parameters again. (actual run), to test the idempotency of the module.
+        * An assert task to verify the object was created successfully for the previous tasks of adding new object with all the parameters. (check mode, actual run and idempotency).
+        * Task to update the object parameters. (check mode) -> where the check_mode is set to true.
+        * Task to update the object parameters. (actual run) -> where the check_mode is set to false.
+        * Task to update the object parameters again. (actual run), to test the idempotency of the module.
+        * An assert task to verify the object was updated successfully for the previous tasks of updating object parameters. (check mode, actual run and idempotency).
+        * Task to query a particular object.
+        * Task to query all objects.
+        * An assert task to verify the object was queried successfully for the previous tasks of querying a particular object and querying all objects.
+        * Task to remove an object. (check mode) -> where the check_mode is set to true.
+        * Task to remove an object. (actual run) -> without check_mode.
+        * Task to remove an object again. (actual run), to test the idempotency of the module.
+        * An assert task to verify the object was removed successfully for the previous tasks of removing an object. (check mode, actual run and idempotency).
+        * Tasks to clean the environment to make sure there are no residual configurations.
```

Comment:
perhaps mention that most of the time the removal of the tenant or the highest parent is enough since this is the parent class in a lot of our cases?